### PR TITLE
Add custom partitioner to streaming tutorial

### LINKED ISSUE #77: Add custom partitioning to the Madoop documentation
Madoop documentation should include information about the new custom partitioning per #71.
This is needed for the Project 5 spec in [this PR](https://github.com/eecs485staff/p5-search-engine/pull/710).

### PR DESCRIPTION
**Pull request by noah-weingarden**
Closes #77. I don't think `README.md` needs any additions because of this after all.

### CODE REVIEW

File: README_Hadoop_Streaming.md [hunk 1]
```diff
@@ -286,6 +286,40 @@ if __name__ == "__main__":
     main()
 ```
 
+## Custom partitioner
```
Comment (awdeorio) [inline README_Hadoop_Streaming.md:289]: I'll suggest moving this section to be the last section.

File: README_Hadoop_Streaming.md [hunk 2]
```diff
@@ -286,6 +286,40 @@ if __name__ == "__main__":
...
+```console
+$ madoop \
+  -input example/input \
+  -output example/output \
+  -mapper example/map.py \
+  -reducer example/reduce.py \
+  -partitioner example/partition.py
+```
+
+This feature works similarly to Hadoop's [`Partitioner` class](https://hadoop.apache.org/docs/current/api/org/apache/hadoop/mapreduce/Partitioner.html). The main difference is that Hadoop only allows partitioners to be a Java class, while Madoop allows any executable that reads from `stdin` and writes to `stdout`.
```
Comment (awdeorio) [inline README_Hadoop_Streaming.md:321]:
```suggestion
This feature is similar to Hadoop's [`Partitioner` class](https://hadoop.apache.org/docs/current/api/org/apache/hadoop/mapreduce/Partitioner.html), although it is not directly compatible with Hadoop. The main difference is that Hadoop only allows partitioners to be a Java class, while Madoop allows any executable that reads from `stdin` and writes to `stdout`.
```

**Review by awdeorio — APPROVED** [review]
LGTM after a couple comments

**Comment by awdeorio** [comment]
Thank you @noah-weingarden !

**Comment by noah-weingarden** [comment]
No problem! :D

**Pull request merged**